py-polars/polars/series/series.py

```diff
@@ -5628,50 +5628,70 @@ def shift(self, n: int = 1, *, fill_value: IntoExpr | None = None) -> Series:
         ]
         """
 
-    def zip_with(self, mask: Series, other: Series) -> Self:
+    def zip_with(self, mask: Series, other: Series | PythonLiteral) -> Self:
         """
-        Take values from self or other based on the given mask.
+        Take values from `self` or `other`, based on the given mask.
 
-        Where mask evaluates true, take values from self. Where mask evaluates false,
-        take values from other.
+        Where `mask` evaluates True, take values from `self`.
+        Where `mask` evaluates False, take values from `other`.
 
         Parameters
         ----------
         mask
             Boolean Series.
         other
-            Series of same type.
+            Series or literal value of the same type.
 
         Returns
         -------
         Series
 
         Examples
         --------
-        >>> s1 = pl.Series([1, 2, 3, 4, 5])
-        >>> s2 = pl.Series([5, 4, 3, 2, 1])
+        Create the mask from a comparison between two Series.
+
+        >>> s1 = pl.Series("a", [1, 2, 3, 4, 5])
+        >>> s2 = pl.Series("b", [5, 4, 3, 2, 1])
         >>> s1.zip_with(s1 < s2, s2)
         shape: (5,)
-        Series: '' [i64]
+        Series: 'a' [i64]
         [
                 1
                 2
                 3
                 2
                 1
         ]
+
+        Supply the mask as a Series of booleans.
+
         >>> mask = pl.Series([True, False, True, False, True])
         >>> s1.zip_with(mask, s2)
         shape: (5,)
-        Series: '' [i64]
+        Series: 'a' [i64]
         [
                 1
                 4
                 3
                 2
                 5
         ]
+
+        Provide `other` as a literal value.
+
+        >>> s1.zip_with(mask, 999)
+        shape: (5,)
+        Series: 'a' [i64]
+        [
+            1
+            999
+            3
+            999
+            5
+        ]
         """
+        if not isinstance(other, Series):
+            other = Series([], dtype=self.dtype).extend_constant(other, n=len(self))
```

Comment:
I don't think this needs to be `length=len(self)`. If it is `length=1`, it will be broadcasts as a scalar and doesn't need to initialize the whole series.

Reply:
Ahh, good catch! Done ✌️